src/main/java/com/arpnetworking/metrics/impl/ApacheHttpSink.java

```diff
@@ -123,10 +130,30 @@ private ApacheHttpSink(final Builder builder) {
     private static final Header CONTENT_TYPE_HEADER = new BasicHeader(HttpHeaders.CONTENT_TYPE, "application/octet-stream");
     private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(ApacheHttpSink.class);
 
+    /* package private */ static final class SingletonSupplier<T> implements Supplier<T> {
+
+        /* package private */ SingletonSupplier(final Supplier<T> supplier) {
+            _supplier = supplier;
+        }
+
+        @Override
+        public synchronized T get() {
```

Comment:
Removed the `AtomicReference` and just went with double check set on a volatile. As discussed, I don't think adding another thread to handle the hand-off from the constructor to the pull makes much sense. In fact, we need to figure something out for clients that would prefer to use a different HTTP dispatch mechanism (e.g. Akka and Vertx).